crates/types/src/line.rs

```diff
@@ -95,10 +106,14 @@ impl<const DIMENSION: usize> Line<DIMENSION> {
         self.squared_distance_to_point(point).sqrt()
     }
 
-    pub fn is_orthogonal(&self, other: &Line<DIMENSION>, epsilon: f32) -> bool {
+    pub fn angle_diff_to_orthogonal(&self, other: &Line<DIMENSION>) -> f32 {
         let self_direction = (self.1 - self.0).normalize();
         let other_direction = (other.1 - other.0).normalize();
-        (self_direction.dot(&other_direction).acos().abs() - PI / 2.0).abs() < epsilon
+        (self_direction.dot(&other_direction).acos().abs() - PI / 2.0).abs()
```

Comment:
You can also check whether two vectors are orthogonal by calculating their dot product and checking if the result is zero. That way you do not need to calculate the angle using `acos`.

Reply:
Now we have code duplication again but it now does not use `acos()` anymore.

Reply:
Careful! You just changed the meaning of the epsilon parameter.
If I see this correctly, you don't even use the `is_orthogonal` function in this PR. It is only used in the field border detection, where an `angle_threshold` parameter is passed for this epsilon.

Furthermore, this implementation isn't even correct.
You just check whether the dot product is less than epsilon without checking for the sign.
If the two vectors point in opposite directions, the normalized dot product would be `-1.0` and the `is_orthogonal` check would return true if we assume epsilon to be some small positive value.

I would suggest reverting this change and renaming the epsilon parameter to something more meaningful.

Reply:
Reverted, now we also don't have the code duplication anymore. Decided against renaming `epsilon`. Should be clear enough. The function also has been renamed to express the absolute return value.

Reply:
I am fine with this version.